### PR TITLE
TILA-1538: Update ApplicationRound, Application and ApplicationEvent statuses

### PR DESCRIPTION
**Pull request by saaste**
This PR removes unused statuses and adds new ones to the following classes:
- `ApplicationRoundStatus`
- `ApplicationStatus`
- `ApplicationEventStatus`

All tests are updated to match the changes and I also wrote new ones to cover the new cases.

I created a separate utility module for `ApplicationRound` status changes. It deals with cases where status of a child entity is updated when `ApplicationRound` status changes.

This was a bit challenging task because of the lack of knowledge of the previous status handling 😅  So take your time to review it. There might be stuff I missed and understood wrong.

### CODE REVIEW

File: applications/models.py [hunk 1]
```diff
@@ -209,19 +209,26 @@ class ApplicationRoundStatus(models.Model, StatusMixin):
     IN_REVIEW = "in_review"
     REVIEW_DONE = "review_done"
     ALLOCATED = "allocated"
+    RESERVING = "reserving"  # kun prosessi on käynnissä
```
Comment (nicobav) [inline applications/models.py:212]: I'd get rid off the finnish comments or translate them to english

File: applications/models.py [hunk 2]
```diff
@@ -209,19 +209,26 @@ class ApplicationRoundStatus(models.Model, StatusMixin):
...
     HANDLED = "handled"
-    VALIDATED = "validated"
-    APPROVED = "approved"
+    SENDING = "sending"  # kun prosessi on käynnissä
```
Comment (nicobav) [inline applications/models.py:214]: :finland:

File: applications/models.py [hunk 3]
```diff
@@ -209,19 +209,26 @@ class ApplicationRoundStatus(models.Model, StatusMixin):
...
+    SENT = "sent"  # mahdollisesti
```
Comment (nicobav) [inline applications/models.py:215]: :finland:

File: applications/utils/status_manager.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+def handle_applications_on_review_done(application_round: ApplicationRound):
+    applications = Application.objects.filter(
+        application_round=application_round,
+        latest_status=ApplicationStatus.IN_REVIEW,
+    )
+    for application in applications:
+        events = ApplicationEvent.objects.filter(
+            application=application,
+        )
+        declined_event_count = len(
```
Comment (nicobav) [inline applications/utils/status_manager.py:33]: If we would change this to queryset it should be a somewhat more efficient;
```
events.filter(latest_status=ApplicationEventStatus.DECLINED).count()
```

File: applications/utils/status_manager.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
+from applications.models import (
+    Application,
+    ApplicationEvent,
+    ApplicationEventStatus,
+    ApplicationRound,
+    ApplicationStatus,
+)
+
+
```
Comment (nicobav) [inline applications/utils/status_manager.py:9]: I notice here a lot of model.status = Something => model.save() 
Should we just call the `set_status` method?

**Review by nicobav — CHANGES_REQUESTED** [review]
Overall looks good :+1:  
Some tiny red marker notes.

File: applications/utils/status_manager.py (hunk 5 above)
Comment (saaste) [reply applications/utils/status_manager.py:9]: Good catch. I also fixed the same thing in tests 👍

File: api/graphql/application_rounds/application_round_types.py [hunk 6]
```diff
@@ -109,7 +108,6 @@ class Meta:
             "application_round_baskets",
             "allocating",
             "aggregated_data",
-            "approved_by",
```
[the remark this answers is not in the document]
Comment (saaste) [reply api/graphql/application_rounds/application_round_types.py:112]: @siren FYI ⚠️

File: applications/allocation_result_mapper.py [hunk 7]
```diff
@@ -45,10 +44,6 @@ def to_events(self):
                     basket_id=allocated_event.basket_id,
                 )
 
-                ApplicationEventStatus.objects.create(
```
Comment (nicobav) [inline applications/allocation_result_mapper.py:48]: This left me pondering what state the application event will get when its gone through the allocation algorithm? Or does it change at all?

**Review by nicobav — CHANGES_REQUESTED** [review]
Some refactor commit cleanups TODO and then we are :rocket:

File: applications/allocation_result_mapper.py (hunk 7 above)
Comment (saaste) [reply applications/allocation_result_mapper.py:48]: I undid the delete. Now we se the event to `APPROVED` status

**Comment by saaste** [comment]
> Some refactor commit cleanups TODO and then we are 🚀

Commits cleaned! 🧹

**Pull request merged**